### PR TITLE
check_rr_graph: fix bug when checking multi-connections between nodes

### PR DESCRIPTION
**Pull request by acomodi**
Signed-off-by: Alessandro Comodi <acomodi@antmicro.com>

#### Description

There seems to be a bug in check_rr_graph related to the check on the multiple connections between RR nodes.

While dealing with https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1711 to fix the RR graph due to the presence of only one node for all the possible sides of a pin, I encountered the VPR error below:

```
Message: in check_rr_graph: node 279 (CHANX) connects to node 149 (IPIN) 2 times - multi-connections only expected for CHAN->CHAN.
```

Given that the comment prior to the check say that this pair of nodes can have multiple connections, I spotted a possible bug in the if statement.

#### Types of changes

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: vpr/src/route/check_rr_graph.cpp [hunk 1]
```diff
@@ -124,16 +124,16 @@ void check_rr_graph(const t_graph_type graph_type,
             t_rr_type to_rr_type = device_ctx.rr_nodes[to_node].type();
 
             /* Only expect the following cases to have multiple edges
-             * - chan <-> chan connections 
-             * - IPIN <-> chan connections (unique rr_node for IPIN nodes on multiple sides)
-             * - OPIN <-> chan connections (unique rr_node for OPIN nodes on multiple sides)
+             * - CHAN <-> CHAN connections
+             * - CHAN  -> IPIN connections (unique rr_node for IPIN nodes on multiple sides)
+             * - OPIN  -> CHAN connections (unique rr_node for OPIN nodes on multiple sides)
```
Comment (litghost) [inline vpr/src/route/check_rr_graph.cpp:129]: My suggestion for clarity is to break this into many lines, e.g.:

```
bool is_chan_to_chan = ...;
bool is_chan_to_ipin = ...;
bool is_opin_to_chan = ...;
```

That will make the intention of each boolean statement clear, and easier to follow.  Then the final if statement will be very simple too!

Comment (acomodi) [reply vpr/src/route/check_rr_graph.cpp:129]: Sure, makes total sense. Done

File: vpr/src/route/check_rr_graph.cpp [hunk 2]
```diff
@@ -124,16 +124,16 @@ void check_rr_graph(const t_graph_type graph_type,
...
              */
-            if (((to_rr_type != CHANX && to_rr_type != CHANY && rr_type != IPIN)
-                 || (rr_type != CHANX && rr_type != CHANY))
-                && ((to_rr_type != CHANX && to_rr_type != CHANY)
-                    || (rr_type != CHANX && rr_type != CHANY && rr_type != OPIN))) {
+            bool is_chan_to_chan = (rr_type == CHANX && to_rr_type == CHANY) || (rr_type == CHANY && to_rr_type == CHANX);
```
Comment (litghost) [inline vpr/src/route/check_rr_graph.cpp:131]: `is_chan_to_chan` should be `(rr_type == CHANX || rr_type == CHANY) && (to_rr_type == CHANX || to_rr_type == CHANY)`.  It is legal to have CHANX <-> CHANX for adjacent tracks.

Comment (acomodi) [reply vpr/src/route/check_rr_graph.cpp]: Oh right, thanks. Fixed

File: vpr/src/route/check_rr_graph.cpp [hunk 3]
```diff
@@ -124,16 +124,16 @@ void check_rr_graph(const t_graph_type graph_type,
             t_rr_type to_rr_type = device_ctx.rr_nodes[to_node].type();
 
             /* Only expect the following cases to have multiple edges
```
Comment (vaughnbetz) [inline vpr/src/route/check_rr_graph.cpp:126]: Suggest a longer comment here.  Here's a proposal.
/* It is unusual to have more than one programmable switch (in the same direction) between a from_node and a to_node, as the duplicate switch doesn't add more routing flexibility. However, such duplicate switches can occur for some types of nodes, which we allow below. Reasons one could have duplicate switches between two nodes include:
 - The two switches have different electrical characteristics.
 - Wires near the edges of an FPGA are often cut off, and the stubs connected together. A regular switch pattern could then result in one physical wire connecting multiple times to other wires, IPINs or OPINs.
*/

**Review by tangxifan — COMMENTED** [review]
Look clean to me.

**Comment by acomodi** [comment]
All CIs went green and I have updated the comment. Merging.

File: vpr/src/route/check_rr_graph.cpp (hunk 3 above)
Comment (acomodi) [reply vpr/src/route/check_rr_graph.cpp]: Done

**Pull request merged**